xla/service/gpu/cudnn_fused_mha_rewriter.cc

```diff
@@ -337,150 +324,128 @@ std::vector<int64_t> GetDimensionVector(absl::Span<const int64_t> dimensions,
   return vec;
 }
 
-absl::StatusOr<bool> IsSupportedBMM1(const HloInstruction* bmm_1,
-                                     bool is_training) {
-  const DotDimensionNumbers& dot_dims_bmm1 = bmm_1->dot_dimension_numbers();
+absl::StatusOr<std::vector<int64_t>> GetBHSD(HloInstruction* bmm_1,
```

Comment:
Maybe return a struct so that you can name the different values that are returned? I think it would make it a bit easier to read the code.

Reply:
I created a QKVLayout struct and use std::optional for checking return status.